overlord/snapstate/handlers.go

```diff
@@ -1753,6 +1780,13 @@ func (m *SnapManager) startSnapServices(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// check what services with "InstallMode: disable" need to be disabled
+	svcsToDisableFromInstallMode, err := installModeDisabledServices(st, snapst, currentInfo)
```

Comment:
I think this needs to take into account also ServicesEnabledByHooks?  is that right @stolowski ?

We need a test where a post-refresh or an install hook enables a service that is install-mode: disable maybe?

Reply:
Yes, you're right, I think the spread test I suggested earlier below would uncover this.

Reply:
I added a unit test and code to handle this but the spread test for this is not done yet

Reply:
I also added a spread test now with a `snapctl start --enable` from the install hook.